### ISSUE TITLE
Navigation to Screen from bottom sheet

### ISSUE THREAD

**Issue by Kashif-E** [adrielcafe/voyager#283]
How can we navigate to a screen from a bottom sheet?
if I do bottomSheetNavigator.show(ScreenA()) and then navigator.push(ScreenB()) it will be another bottom sheet. How can i do the opposite?

**Comment by DevSrSouza** [comment]
Hey, sorry the delay!

Don't know if I understand your question properly, do you want to navigate to another Screen inside the BottomSheet or you want  to navigate to Screen of the a Navigator that is the Content of the Scaffold? Can you give me some examples?

**Comment by Kashif-E** [comment]
Hey! Sorry for the delay, consider this glow when moving from a bottom sheet to a non-bottom sheet screen. 
![ezgif-2-b460334530](https://github.com/adrielcafe/voyager/assets/61690178/22e6e76e-021f-46dc-b8d4-a9efedb35ae4)

**Comment by akhbulatov** [comment]
@Kashif-E same problem, have you found any solution?

**Comment by akhbulatov** [comment]
@DevSrSouza is there a recommended solution for such a case?

**Comment by mrcsxsiq** [comment]
Same here. Does anyone have some solutions?

**Comment by sausti** [comment]
👋 We're also looking for a solution to this. The structure we adopt is the root composable of our application hosts the `BottomSheetNavigator`. For example:
```
@Composable
fun App() {
    BottomSheetNavigator {
        Navigator(screen = MainScreen())
    }
}
```
When in a bottom sheet screen, `LocalNavigator.currentOrThrow` is thus the `BottomSheetNavigator`, which explains why all screens launched from this context open as bottom sheets. `LocalNavigator.currentOrThrow.parent` is, of course, null.

I _think_  `BottomSheetNavigator` needs a variant where  `BottomSheetNavigatorContent` is a `Navigator`, where such `Navigator` is made available via a `CompositionLocal`?

**Comment by Kashif-E** [comment]
you can pass a navigator to the bottom sheet and then use that inside the bottom sheet to open a new screen instead of the bottom sheet

**Comment by mrcsxsiq** [comment]
> you can pass a navigator to the bottom sheet and then use that inside the bottom sheet to open a new screen instead of the bottom sheet

This solution worked for me. Thank you @Kashif-E

**Comment by akardas16** [comment]
> you can pass a navigator to the bottom sheet and then use that inside the bottom sheet to open a new screen instead of the bottom sheet

if you put the app in the background while bottom sheet is open, app will crash because of Parcelable IOException. I can repeat the crash continuously.

**Comment by Kashif-E** [comment]
@akardas16  yes you are right, the navigator is not parcelable/serializable. Another option is to pass a result to the original bottom sheet caller; upon receiving the result, you can navigate to the next screen. you can use one of the ways mentioned in this repo for result passing.

navigate to the bottom sheet  from screen A-> hide the bottom sheet with the result -> get the result  in screen A and navigate to screen B

**Comment by akardas16** [comment]
> @akardas16 yes you are right, the navigator is not parcelable/serializable. Another option is to pass a result to the original bottom sheet caller; upon receiving the result, you can navigate to the next screen. you can use one of the ways mentioned in this repo for result passing.
> 
> navigate to the bottom sheet from screen A-> hide the bottom sheet with the result -> get the result in screen A and navigate to screen B

I have tried a few options which mentioned but looks like they are not working as expected. there is no reason to not using **ModalBottomSheet** if needs result from bottom sheet.

**Comment by Kashif-E** [comment]
Can you explain?
